File: LeetCode/Java/1464. Maximum Product of Two Elements in an Array #1.java
class Solution {
    public int maxProduct(int[] nums) {
        int n = nums.length;
        int max = Integer.MIN_VALUE;
        
        for (int i = 0; i < n; i++)
            for (int j = i + 1; j < n; j++)
                max = Math.max(max, (nums[i] - 1) * (nums[j] - 1));
        
        return max;
    }
}

File: LeetCode/Python/1464. Maximum Product of Two Elements in an Array #1.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        n = len(nums)
        _max = -float("inf")
        
        for i in range(n):
            for j in range(i + 1, n):
                _max = max(_max, (nums[i] - 1) * (nums[j] - 1))
        
        return _max
